generator/src/main/java/org/apache/kafka/message/SchemaGenerator.java

```diff
@@ -317,8 +317,11 @@ private String fieldTypeToSchemaType(FieldType type,
                         fieldTypeToSchemaType(arrayType.elementType(), false, version, fieldFlexibleVersions, false));
             }
         } else if (type.isStruct()) {
-            return String.format("%s.SCHEMA_%d", type,
+            if (nullable)
+                headerGenerator.addImport(MessageGenerator.NULLABLE_SCHEMA_CLASS);
+            String schemaType = String.format("%s.SCHEMA_%d", type,
                 floorVersion(type.toString(), version));
+            return nullable ? String.format("new NullableSchema(%s)", schemaType) : schemaType;
```

Comment:
This generates the following code. Should we just change Assignment.SCHEMA_0 to be NullableSchema?

 `           new Field("assignment", new NullableSchema(Assignment.SCHEMA_0), "null if not provided; the assignment otherwise."),`